commit/solvers.py

```diff
@@ -1,157 +1,605 @@
 import numpy as np
+from math import sqrt
+from warnings import warn
+from commit.proximals import fista, omega_hierarchical, prox_hierarchical
 import sys
-
 eps = np.finfo(float).eps
 
+def issolver(solver=None):
+    """
+    Check if this version of COMMIT is endowed with the wanted solver.
 
-def nnls( y, A, At = None, tol_fun = 1e-4, tol_x = 1e-9, max_iter = 100, verbose = 1, x0 = None ) :
-    """Solve non negative least squares problem of the following form:
+    Parameters
+    ----------
+    solver: string (default=None)
+        name of the solver
 
-       min 0.5*||y-A x||_2^2 s.t. x >= 0
+    Returns
+    ----------
+    Boolean if solver is a non-empty string, otherwise it returns the list of
+    available solvers
 
-    The problem is solved using the forward-backward algorithm with FISTA-like acceleration.
+    Notes
+    -----
+    Author: Matteo Frigo - athena @ INRIA
```

Comment:
This may sound stupid, but we should find the good way to (1) acknowledge the work of Rafa and (2) the fact that this part was done during your master thesis in Verona/EPFL

Reply:
True. I could write `Matteo Frigo - ATHENA @ Inria, LTS5 @ EPFL, Dep. of CS @ Verona` for my side and put Rafa as first author of the fista routine.